src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationUnitCompletedEvent.cs

```diff
@@ -11,6 +12,11 @@ internal sealed class CompilationUnitCompletedEvent : CompilationEvent
         public CompilationUnitCompletedEvent(Compilation compilation, SyntaxTree compilationUnit, TextSpan? filterSpan = null)
             : base(compilation)
         {
+            if (!compilation.ContainsSyntaxTree(compilationUnit))
```

Comment:
Are we changing behavior here in order to have more of a "fail-fast" experience?

Reply:
That will be the result, but the main reason I changed it here instead of inline was because it was easier to assert here.

Reply:
I think this can be a Debug assert instead of a runtime check. There is no public API entry point to this constructor.